docs/notebooks/create_synthetic_data_from_gretel_dgan_api.ipynb

```diff
@@ -0,0 +1,321 @@
+   "outputs": [],
+   "source": [
+    "# Setup config and train model\n",
+    "\n",
+    "TMP_FILE = \"tmp_train.csv\"\n",
+    "\n",
+    "CONFIG_STRING = f\"\"\"\n",
+    "schema_version: 1.0\n",
+    "\n",
+    "name: \"oildata\"\n",
+    "\n",
+    "models:\n",
+    "  - timeseries_dgan:\n",
+    "        data_source: \"_\"\n",
+    "\n",
+    "        time_column: \"Date\"\n",
+    "        example_id_column: \"attributes\"\n",
+    "        df_style: \"long\"\n",
+    "        \n",
+    "        params:\n",
+    "            epochs: 10\n",
+    "            max_sequence_len: {seq_len}\n",
+    "            sample_len: {seq_len}  # Must evenly divide max_sequence_len, length of time series\n",
+    "            batch_size: 5000 \n",
+    "            generator_learning_rate: 0.0001\n",
+    "            discriminator_learning_rate: 0.0001\n",
+    "            attribute_discriminator_learning_rate: 1e-4\n",
+    "            apply_feature_scaling: True\n",
+    "            apply_example_scaling: True\n",
+    "            feature_num_layers: 3\n",
+    "            feature_num_units: 100\n",
+    "            feature_noise_dim: 10\n",
+    "            \n",
+    "        generate:\n",
+    "            num_records: 500\n",
+    "\n",
+    "\"\"\"\n",
+    "config = yaml.safe_load(CONFIG_STRING)\n",
+    "\n",
+    "project = get_project(display_name=\"DGAN\", create=True)\n",
+    "\n",
+    "print(f\"Follow model training at: {project.get_console_url()}\")\n",
+    "\n",
+    "model = project.create_model_obj(model_config=config)\n",
+    "\n",
+    "df_input.to_csv(TMP_FILE, index=False)\n",
+    "model.data_source = TMP_FILE\n",
+    "\n",
+    "model.submit(upload_data_source=True)\n",
+    "\n",
+    "poll(model)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3133e5f3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Grab synthetic data\n",
+    "\n",
+    "synthetic_df = pd.read_csv(model.get_artifact_link(\"data_preview\"), compression=\"gzip\")\n",
+    "synthetic_df = synthetic_df.drop(columns = 'attributes')\n",
+    "synthetic_df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1bfac4a5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's compare the correlations in the synthetic data between the variables and the correlations in the real data between the variables. \n",
+    "# We want to see that all the cells are as close to 0 as possible.\n",
+    "print(\"Difference in real correlations and synethic data correlations:\")\n",
+    "print(df.iloc[: , 1:].corr() - synthetic_df.iloc[: , 1:].corr())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0a3d8f5f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Now, let's visualize the probability distribution of each feature and it's respective synthetic data alternate.\n",
+    "for val in list(df.iloc[:,1:].columns):\n",
```

Comment:
Minor - This gives a warning,
/usr/local/lib/python3.7/dist-packages/numpy/core/fromnumeric.py:3208: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  return asarray(a).size
/usr/local/lib/python3.7/dist-packages/matplotlib/cbook/__init__.py:1376: VisibleDeprecationWarning: Creating an ndarray from ragged nested sequences (which is a list-or-tuple of lists-or-tuples-or ndarrays with different lengths or shapes) is deprecated. If you meant to do this, you must specify 'dtype=object' when creating the ndarray.
  X = np.atleast_1d(X.T if isinstance(X, np.ndarray) else np.asarray(X))